Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -164,6 +182,25 @@ + (void)clearConfiguration
     globalConfiguration() = nil;
 }
 
++ (void)maybeConsumeBundlePaths:(NSDictionary<NSAttributedStringDocumentReadingOptionKey, id> *)options
+{
+    auto* readAccessPaths = dynamic_objc_cast<NSArray<NSURL *>>(options[WKReadAccessFileURLsOption]);
+    if (!readAccessPaths)
+        return;
+
+    if (!readAccessPaths.count)
+        return;
+
+    if (readAccessPaths.count > maximumReadOnlyAccessPaths)
+        [NSException raise:NSInvalidArgumentException format:@"WKReadAccessFileURLsOption may have at most two additional directories."];
+
+    if ([readAccessPaths isEqualToArray:self.readOnlyAccessPaths])
+        return;
+
+    [self.readOnlyAccessPaths setArray:readAccessPaths];
```

Comment:
If you want to verify that `readAccessPaths` only contains `NSURL` objects (and it seems you do), you should write a `for()` loop to iterate through the objects here to check for that _before_ this code.  Something like this:

```
    for (id url in readOnlyAccessPaths) {
        if (!dynamic_objc_cast<NSURL>(url))
            return;
    }
```

Reply:
Will do.

Reply:
I wouldn’t use dynamic_objc_cast just for a type check. If we want syntactic sugar for that we should add another named function. I would write `if (![url isKindOfClass:NSURL.class])`